pkg/tempo/config.go

```diff
@@ -115,8 +116,16 @@ func (c *InstanceConfig) otelConfig() (*configmodels.Config, error) {
 		}
 	}
 
+	switch c.PushConfig.Compression {
+	case "none":
+		c.PushConfig.Compression = ""
```

Comment:
Mutating the config feels a little strange; can we have this set a separate variable instead?